### PR TITLE
[SPARK-49028][CONNECT][SQL] Create a shared SparkSession

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?
This PR creates a common base class for `SparkSession`.

### Why are the changes needed?
We are creating a unified interface for the Scala SQL Client API for Classic and Connect.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Compilation and existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala
Comment (hvanhovell) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:1]: Use `@inheritdoc` where possible.

File: sql/api/src/main/scala/org/apache/spark/sql/api/SparkSession.scala
Comment (hvanhovell) [inline sql/api/src/main/scala/org/apache/spark/sql/api/SparkSession.scala:1]: Use `@inheritdoc` where possible.

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala
Comment (hvanhovell) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala:1]: Use `@inheritdoc` where possible.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by hvanhovell** [comment]
Merging to master

**Pull request closed without merging**